### PR TITLE
Add github app and metadata to benchmark json.

### PR DESCRIPTION
**Pull request by gs0510**

### CODE REVIEW

File: pipeline.ml [hunk 1]
```diff
@@ -12,118 +12,144 @@ let read_fpath p = Bos.OS.File.read p |> Rresult.R.error_msg_to_invalid_arg
...
+        Github.Api.Repo.ci_refs repo
+        |> Current.list_iter ~pp:Github.Api.Commit.pp @@ fun head ->
+           let src = Git.fetch (Current.map Github.Api.Commit.id head) in
+           let image = Docker.build ~pool ~pull:false ~dockerfile (`Git src) in
+           let docker_cpuset_cpus =
+             match docker_cpu with
+             | Some i -> [ "--cpuset-cpus"; string_of_int i ]
+             | None -> []
+           in
+           let docker_cpuset_mems =
```
Comment (MagnusS) [inline pipeline.ml]: Would it be possible to leave these helper functions one level up? They don't seem to change the value in each iteration and would make it a bit easier to see what's new

File: pipeline.ml [hunk 2]
```diff
@@ -12,118 +12,144 @@ let read_fpath p = Bos.OS.File.read p |> Rresult.R.error_msg_to_invalid_arg
 let read_channel_uri p =
   read_fpath p |> String.trim |> Uri.of_string |> Current_slack.channel
 
+let pool = Current.Pool.create ~label:"docker" 1
+
+let get_commit ~head =
+  let { Github.Repo_id.owner; name } = Github.Api.Commit.repo_id head in
+  let hash = Github.Api.Commit.hash head in
+  let res = hash ^ owner ^ name in
+  res
```
Comment (CraigFe) [inline pipeline.ml]: Same as just having `hash ^ owner ^ name` as the last line, but fine if you're deliberately naming it.

File: pipeline.ml [hunk 3]
```diff
@@ -12,118 +12,144 @@ let read_fpath p = Bos.OS.File.read p |> Rresult.R.error_msg_to_invalid_arg
...
 let dockerfile ~base =
   let open Dockerfile in
   from (Docker.Image.hash base)
   @@ run
        "sudo apt-get install -qq -yy libffi-dev liblmdb-dev m4 pkg-config \
-        gnuplot-x11"
+        gnuplot-x11 capnproto libcapnp-dev libgmp-dev libsqlite3-dev"
   @@ copy ~src:[ "--chown=opam:opam ." ] ~dst:"index" ()
   @@ workdir "index"
+  @@ run "ls"
```
Comment (CraigFe) [inline pipeline.ml]: Are these `ls`s in here for debugging purposes?

File: pipeline.ml (hunk 1 above)
Comment (gs0510) [reply pipeline.ml]: Yep will do!

File: pipeline.ml (hunk 3 above)
Comment (gs0510) [reply pipeline.ml]: yarp, will remove!

**Comment by gs0510** [comment]
> It looks as though this PR gives up the ability of running the benchmarks independently of a GitHub webhook trigger / App configuration and such. Personally, I find it useful to work independently of that API when making changes to the pipeline and testing (this was the motivation behind PRs #3 and #4).
> 
> If you think it's not worth maintaining this code purely for testing purposes, that's OK. Otherwise, we can look at using `Cmdliner` subcommands (or required options) to allow toggling the input mode at runtime.

Sorry I just saw this! You raise an excellent point! How would a Cmdliner subcommand help with a GithubApp though, at the moment it is very tightly coupled with the pipeline steps...

**Comment by CraigFe** [comment]
Both versions of the pipeline at some point depend on a `src`, after which they behave in the same way. This could be extracted to an internal `src Current.t -> unit Current.t` pipeline segment. 

We would then split the runtime API into two endpoints, consuming `Github.App.t` and `Fpath.t` respectively. The GitHub endpoint gets a list of sources and creates one pipeline segment for each one; the local endpoint gets a single source and creates one pipeline segment.

Finally, change the `Cmdliner` interface to expose two subcommands: one of which constructs a `Github.App.t` and invokes the GitHub endpoint

**Pull request closed without merging**